core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -83,6 +83,17 @@ package object config {
   private[spark] val EXECUTOR_CLASS_PATH =
     ConfigBuilder(SparkLauncher.EXECUTOR_EXTRA_CLASSPATH).stringConf.createOptional
 
+  private[spark] val EXECUTOR_HEARTBEAT_DROP_ZERO_METRICS =
+    ConfigBuilder("spark.executor.heartbeat.dropZeroMetrics").booleanConf.createWithDefault(true)
```

Comment:
Question -- when would you not want this to be true? It's already changing behavior here, but what's the case where you need a safety valve to go back? it's just not broadcasting changes that can't matter because they're zero?

Reply:
OK, it's an internal-only safety-valve flag.